### PR TITLE
Uses a terms array to query data

### LINKED ISSUE #698: Use `terms` instead of `term`
## Problem

In the search controller, when performing filtering for more than one value, we use an array of `term` queries instead of using `terms`. 

https://github.com/WordPress/openverse-api/blob/6cc0551c639081f895d974b30cc465713d0ce360/api/catalog/api/controllers/search_controller.py#L172-L177

## Description
Replace the array of `term` queries with a singular `term` query. According to the [ES docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-terms-query.html), the `terms` query is functionally identical to `term`.

## Additional context

For an example, a `terms` query is used just a few lines below in the same file, inside the `_exclude_filtered` function.

https://github.com/WordPress/openverse-api/blob/6cc0551c639081f895d974b30cc465713d0ce360/api/catalog/api/controllers/search_controller.py#L197

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by ramadanomar**
## Fixes
#698 by @dhruvkb 

## Description
Sends the query arguments as an array. All credit goes to @sarayourfriend  and @obulat for their code on #764 pr. I tested it out on POSTMAN and it works perfectly. 

image

As you can see, the server recieves query parameters as an array

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by ramadanomar** [comment]
I'm not sure on how i can add @sarayourfriend and @obulat as authors/ co-authors of this pr. I would greatly appreciate if someone can edit the pr to include them as they deserve all of the credits.

**Review by AetherUnbound — COMMENTED** [review]
This looks good! I was able to run some queries locally which behaved as expected.

It looks like linting is failing though.

One thing you could look into if you wanted to credit Sara and Olga is [GitHub's co-authored-by](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) 😄

**Comment by ramadanomar** [comment]
Oops , didn't mean to remove the review request from @obulat

**Review by AetherUnbound — APPROVED** [review]

**Review by AetherUnbound — APPROVED** [review]

File: api/catalog/api/controllers/search_controller.py [hunk 1]
```diff
@@ -241,15 +241,14 @@ def _apply_filter(
...
+        if arguments is None:
+            return s
+        arguments = arguments.split(",")
+        parameter = es_field or serializer_field
+        query = Q("terms", **{parameter: arguments})
         method = getattr(s, behaviour)
-        return method("bool", should=filters)
-    else:
-        return s
+        return method("bool", should=query)
```
Comment (krysal) [inline api/catalog/api/controllers/search_controller.py:252]:
```suggestion
        return method("bool", should=query)

    return s
```

File: api/catalog/api/controllers/search_controller.py [hunk 2]
```diff
@@ -241,16 +241,16 @@ def _apply_filter(
...
 
+    return s
```
Comment (krysal) [inline api/catalog/api/controllers/search_controller.py:254]:
```suggestion
    return s


```

**Review by krysal — APPROVED** [review]
Nice! Love how this looks much clearer :) Thank you @ramadanomar!

**Pull request merged**